boards/DCM/Src/Io/Io_LSM6DS33.c

```diff
@@ -0,0 +1,214 @@
+ * @return The status indicated by the HAL library for the I2c required
+ * operations
+ */
+HAL_StatusTypeDef Io_LSM6DS33_WriteToImu(
+    uint16_t register_address,
+    uint8_t *data,
+    uint16_t data_size)
+{
+    return HAL_I2C_Mem_Write(
+        lsm_6ds33.imu_i2c_handle, IMU_WRITE_ADDR, register_address,
+        I2C_MEMADD_SIZE_8BIT, data, data_size, 1000);
+}
+
+/**
+ * Converts acceleration values from the IMU into m/s^2
+ * @param accel_from_imu
+ * @return
+ */
+static float Io_LSM6DS33_ConvertIMUAccelerationToMetersPerSecondSquared(
+    int16_t accel_from_imu)
+{
+    const float max_abs_g_force           = 2.0f;
+    const float max_abs_raw_accel_reading = 1 << 15;
+
+    return accel_from_imu * max_abs_g_force / max_abs_raw_accel_reading * 9.8f;
+}
+
+/**
+ * Read data from the IMU registers
+ * @param read_start_address The address on the IMU to start reading from
+ * @param data The array to save the read data into
+ * @param data_size The size of the array (in # of bytes)
+ * @return The status indicated by the HAL library for the I2c required
+ * operations
+ */
+HAL_StatusTypeDef Io_LSM6DS33_ReadFromImu(
+    uint16_t read_start_address,
+    uint8_t *data,
+    uint16_t data_size)
+{
+    return HAL_I2C_Mem_Read(
+        lsm_6ds33.imu_i2c_handle, IMU_READ_ADDR, read_start_address,
+        I2C_MEMADD_SIZE_8BIT, data, data_size, 100);
+}
+
+HAL_StatusTypeDef Io_LSM6DS33_ConfigureImu(I2C_HandleTypeDef *i2c_handle)
+{
+    lsm_6ds33.imu_i2c_handle = i2c_handle;
+    HAL_StatusTypeDef status = HAL_OK;
+
+    //===== Initialize the IMU chip =======
+
+    // Wait until the device is ready to communicate
+    status =
+        HAL_I2C_IsDeviceReady(lsm_6ds33.imu_i2c_handle, IMU_READ_ADDR, 3, 1000);
+
+    // Enable Accelerometer X,Y,Z axis
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x38; // 0b00_111_000
+        status       = Io_LSM6DS33_WriteToImu(CTRL9_XL, &data, 1);
+    }
+
+    // Set Accelerometer to read at 416hz, max acceleration to +/-2g,
+    // anti-aliasing filter bandwidth at 400hz
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x60; // 0b0110_00_00
+        status       = Io_LSM6DS33_WriteToImu(CTRL1_XL, &data, 1);
+    }
+
+    // Enable Gyroscope X,Y,Z axis
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x38; // 0b00_111_0_0_0
+        status       = Io_LSM6DS33_WriteToImu(CTRL10_C, &data, 1);
+    }
+
+    // Set Gyroscope to read at 416hz, with max expected rotation of 250dps
+    // (degrees per second)
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x60; // 0b0110_00_0_0
+        status       = Io_LSM6DS33_WriteToImu(CTRL2_G, &data, 1);
+    }
+
+    // Set IMU to trigger an interrupt on INT2 whenever there is new
+    // gyroscope or accelerometer data available
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x3; // 0b00000011
+        status       = Io_LSM6DS33_WriteToImu(INT2_CTRL, &data, 1);
+    }
+
+    initialized = (status == HAL_OK);
```

Comment:
```suggestion
```

Reply:
Why?

Reply:
Having a bool to check whether the IO code has been initialized is not something that is done for the current IO code.

Reply:
Ok. I'll just remove the checks entirely then